src/components/VideoPlayerContexts/PlaybackContext.tsx

```diff
@@ -58,25 +97,43 @@ function PlaybackContextProvider({children}: ChildrenProps) {
      */
     const updateCurrentPlayingReportID = useCallback(
         (state: NavigationState) => {
-            if (!isReportTopmostSplitNavigator()) {
-                setCurrentReportID(undefined);
-                return;
+            const focusedRoute = findFocusedRoute(state);
+            let {reportID} = getReportOrDraftReport(Navigation.getTopmostReportId()) ?? {};
+
+            // We need to handle a case where a report is selected via search and is therefore not a topmost report,
+            // but we still want to be able to play videos in it
+            if (isMoneyRequestReportRouteWithReportIDInParams(focusedRoute)) {
+                reportID = focusedRoute.params.reportID;
             }
-            const reportID = Navigation.getTopmostReportId(state);
+
+            reportID = Navigation.getActiveRouteWithoutParams() === `/${ROUTES.ATTACHMENTS.route}` ? prevCurrentReportID : reportID;
+
             setCurrentReportID(reportID);
         },
-        [setCurrentReportID],
+        [setCurrentReportID, prevCurrentReportID],
     );
 
     const updateCurrentlyPlayingURL = useCallback(
         (url: string | null) => {
             if (currentlyPlayingURL && url !== currentlyPlayingURL) {
                 pauseVideo();
             }
-            setCurrentlyPlayingURLReportID(currentReportID);
+
+            // Used for /attachment route
+            const topMostReport = getReportOrDraftReport(Navigation.getTopmostReportId());
+            const reportIDFromUrlParams = new URLSearchParams(Navigation.getActiveRoute()).get('reportID') ?? undefined;
```

Comment:
I think all of these should have URL uppercase in name based on our style guides
```suggestion
            const reportIDFromURLParams = new URLSearchParams(Navigation.getActiveRoute()).get('reportID') ?? undefined;
```